framework/web/Request.php

```diff
@@ -1508,7 +1509,8 @@ protected function loadCookies()
             }
         } else {
             foreach ($_COOKIE as $name => $value) {
-                $cookies[$name] = new Cookie([
+                $cookies[$name] = Yii::createObject([
+                    'class' => 'yii\web\Cookie',
```

Comment:
in the two cases above it is created from data, should we really allow modifying these?

Reply:
Modifying data isn't the only case. You may set your own implementation as well. Not sure it's a good idea to use container in one place but not use it in another.